ArmoniK.Extension.CSharp.DllWorker/LibraryLoader.cs

```diff
@@ -0,0 +1,297 @@
+// 
+// Copyright (C) ANEO, 2021-2024. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Concurrent;
+using System.IO.Compression;
+using System.Reflection;
+using System.Runtime.Loader;
+
+using ArmoniK.Api.Worker.Worker;
+using ArmoniK.Extension.CSharp.DllCommon;
+using ArmoniK.Extension.CSharp.Worker;
+
+namespace ArmoniK.Extension.CSharp.DllWorker;
+
+public class LibraryLoader : ILibraryLoader
+{
+  private ConcurrentDictionary<string, (Assembly assembly, AssemblyLoadContext loadContext)> assemblyLoadContexts_ = new();
+
+  public ILogger Logger;
+
+  public LibraryLoader(ILoggerFactory loggerFactory)
+    => Logger = loggerFactory.CreateLogger<LibraryLoader>();
+
+  public void Dispose()
+    => assemblyLoadContexts_ = new ConcurrentDictionary<string, (Assembly, AssemblyLoadContext)>();
+
+  public AssemblyLoadContext GetAssemblyLoadContext(string libraryContextKey)
+  {
+    var exists = assemblyLoadContexts_.TryGetValue(libraryContextKey,
+                                                   out var value);
+    if (!exists)
+    {
+      Logger.LogError($"AssemblyLoadContexts does not have key {libraryContextKey}");
+      throw new WorkerApiException("No key found on AssemblyLoadContexts dictionary");
+    }
+
+    return value.loadContext;
+  }
+
+  public void ResetService()
+    => Dispose();
+
+  public async Task<string> LoadLibrary(ITaskHandler      taskHandler,
+                                        CancellationToken cancellationToken)
+  {
+    try
+    {
+      Logger.LogInformation("Starting to LoadLibrary");
+      Logger.LogInformation($"Nb of current loaded assemblies: {assemblyLoadContexts_.Count}");
+
+      var taskLibrary = taskHandler.TaskOptions.GetServiceLibrary();
+
+      //Get the data about the dynamic library
+      var dynamicLibrary = taskHandler.TaskOptions.GetTaskLibraryDefinition(taskLibrary);
+      var filename       = $"{dynamicLibrary}.zip";
+
+      var filePath = @"/tmp/zip";
+
+      var destinationPath = @"/tmp/assemblies";
+
+      var pathToDllFile = dynamicLibrary.PathToFile;
+
+      var dllFileName = dynamicLibrary.DllFileName;
+
+      Logger.LogInformation($"Starting Dynamic loading - TaskLibrary: {taskLibrary}, FileName: {filename}, FilePath: {filePath}, DestinationToUnZip:{destinationPath}, PathToDllFile:{pathToDllFile}, DllFileName: {dllFileName}, Namespace: {dynamicLibrary.Namespace}, Service: {dynamicLibrary.Service}");
```

Comment:
you should not use $ in logger. Consider using template approach as it will create logs easily searchable https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-message-template